tests/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFileTest.java

```diff
@@ -0,0 +1,151 @@
+// Copyright 2024 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.github.bazelbuild.rules_jvm_external.resolver.lockfile;
+
+import static com.google.common.base.StandardSystemProperty.USER_HOME;
+import static org.junit.Assert.assertEquals;
+
+import com.github.bazelbuild.rules_jvm_external.Coordinates;
+import com.github.bazelbuild.rules_jvm_external.resolver.Conflict;
+import com.github.bazelbuild.rules_jvm_external.resolver.DependencyInfo;
+import com.google.gson.GsonBuilder;
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeMap;
+import org.junit.Test;
+
+public class V2LockFileTest {
+
+  private final URI defaultRepo = URI.create("http://localhost/m2/repository/");
+  private final Set<URI> repos = Set.of(defaultRepo);
+
+  @Test
+  public void shouldRenderAggregatingJarsAsJarWithNullShasum() {
+    DependencyInfo aggregator =
+        new DependencyInfo(
+            new Coordinates("com.example:aggregator:1.0.0"),
+            repos,
+            Optional.empty(),
+            Optional.empty(),
+            Set.of(),
+            Set.of(),
+            new TreeMap<>());
+
+    Map<String, Object> rendered = new V2LockFile(repos, Set.of(aggregator), Set.of()).render();
+
+    Map<?, ?> artifacts = (Map<?, ?>) rendered.get("artifacts");
+    Map<?, ?> data = (Map<?, ?>) artifacts.get("com.example:aggregator");
+    Map<?, ?> shasums = (Map<?, ?>) data.get("shasums");
+
+    HashMap<Object, Object> expected = new HashMap<>();
+    expected.put("jar", null);
+    assertEquals(expected, shasums);
+  }
+
+  @Test
+  public void shouldRoundTripASimpleSetOfDependencies() {
+    V2LockFile roundTripped = roundTrip(new V2LockFile(repos, Set.of(), Set.of()));
+
+    assertEquals(repos, roundTripped.getRepositories());
+    assertEquals(Set.of(), roundTripped.getDependencyInfos());
+    assertEquals(Set.of(), roundTripped.getConflicts());
+  }
+
+  @Test
+  public void shouldRoundTripM2Local() {
+    V2LockFile lockFile = new V2LockFile(repos, Set.of(), Set.of());
+    Map<String, Object> rendered = lockFile.render();
+    rendered.put("m2local", true);
+
+    V2LockFile roundTripped =
+        V2LockFile.create(
+            new GsonBuilder().setPrettyPrinting().serializeNulls().create().toJson(rendered));
```

Comment:
Use the `roundTrip` helper?

Reply:
We amended the generated JSON by hand by adding `m2local` to it, so we can't use the `roundTrip` helper here. We _could_ have added the `M2_LOCAL_URI` to the list of repos when we create the lockfile instance, but I don't wanted to be sure we were honouring the `m2local` value in the rendered map properly.